### PR TITLE
Add test cases to check native JSON support for Foreach mediator

### PR DESCRIPTION
**Pull request by arunans23**
Ports https://github.com/wso2/product-ei/pull/4323 and adds additional testcases

### CODE REVIEW

**Comment by arunans23** [comment]
Merge this after https://github.com/wso2/wso2-synapse/pull/1590 is merged and synapse is updated

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+public class ForeachNativeJSONTestCase extends ESBIntegrationTest {
+    private JsonParser parser;
+
+    @BeforeClass(alwaysRun = true)
+    public void uploadSynapseConfig() throws Exception {
+        super.init();
+        setJsonPathConfiguration();
+        loadESBConfigurationFromClasspath(File.separator + "artifacts" + File.separator + "ESB" +
+                File.separator + "mediatorconfig" + File.separator + "foreach" +
+                File.separator + "foreach_native_json_configurations.xml");
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java:59]: do we need these lines

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                File.separator + "jsonArrayAsRootInput.json"));
+        executeSequenceAndAssertResponse("foreachsample", "/foreachjson3", jsonArrayAsRootInput, expectedOutput,
+                "Foreach mediator couldn't process Json array when it is there as root");
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void stop() throws Exception {
+        super.cleanup();
+    }
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java:155]: this is also of no use

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java (hunk 1 above)
Comment (arunans23) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java]: Fixed

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java (hunk 2 above)
Comment (arunans23) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForeachNativeJSONTestCase.java]: Fixed

**Pull request merged**